api/model.go

```diff
@@ -1860,8 +1924,9 @@ func toBackingImageResource(bi *longhorn.BackingImage, apiContext *api.ApiContex
 		DeletionTimestamp: deletionTimestamp,
 	}
 	res.Actions = map[string]string{
-		"backingImageCleanup": apiContext.UrlBuilder.ActionLink(res.Resource, "backingImageCleanup"),
-		BackingImageUpload:    apiContext.UrlBuilder.ActionLink(res.Resource, BackingImageUpload),
+		"backingImageCleanup":      apiContext.UrlBuilder.ActionLink(res.Resource, "backingImageCleanup"),
+		BackingImageUpload:         apiContext.UrlBuilder.ActionLink(res.Resource, BackingImageUpload),
+		"backupBackingImageCreate": apiContext.UrlBuilder.ActionLink(res.Resource, "backupBackingImageCreate"),
```

Comment:
`BackingImageUpload` is `upload` which is different from others. Is this intended? It seems the prefix already exists, should we just name `cleanup`, `upload` and `create` instead?

Reply:
I can fix that and notice UI to adjust as well

Reply:
by the way, this is under backingimage
but for create, it is for creating BackupBackingImage, I am not sure if it will confuse others if we only use "create"

Reply:
I think adding prefix is more appropriate.
We put all the resource and their actions in the same file `model.go`, for each action they also have Input
Adding prefix can distinguish them

e.g.

```
type BackingImageCleanupInput struct {
	Disks []string `json:"disks"`
}

type PVCreateInput struct {
	PVName string `json:"pvName"`
	FSType string `json:"fsType"`

	SecretName      string `json:"secretName"`
	SecretNamespace string `json:"secretNamespace"`

	StorageClassName string `json:"storageClassName"`
}

type PVCCreateInput struct {
	Namespace string `json:"namespace"`
	PVCName   string `json:"pvcName"`
}
```

Reply:
> `BackingImageUpload` is `upload` which is different from others. Is this intended? It seems the prefix already exists, should we just name `cleanup`, `upload` and `create` instead?

I think we just forgot to make it consistent to have prefix when implementing
https://github.com/ChanYiLin/longhorn-manager/commit/f271ef470f706e58cbd0d30948e1f1189b028d8d#diff-cb740c237beef9090d8b3a15ca8b76aac83a3af2d52deadc382b4309916a715cL141

I can fix that, but need UI/backing-image-manager changes or it will break the API
And also need to check harvester side.

Reply:
I see why do we use "upload" without prefix

1. in `model.go` we should use the prefix because we put all the resources in the same places, so we have "BackingImageCleanup"
2. When implementing "upload", since it is a proxy from `longhorn-manager` -> `backing-image-manager/datasource` -> `backing-image-manager/sync-file`

https://github.com/longhorn/backing-image-manager/blob/5c9b6e5c6f3ad1d24e84a344462d6a434dd48f98/pkg/datasource/router.go#L8-L19

https://github.com/longhorn/backing-image-manager/blob/5c9b6e5c6f3ad1d24e84a344462d6a434dd48f98/pkg/sync/router.go#L22-L27

3. we proxy the "action" which is "upload" all the way down to the `sync-file server`. Since we added the `proxy` after `datasource` and `sync-file` so we keep using the "upload".

To fix the wording we have to change
- datasource in backing-image-manager, but that make the action name there not consistent to others which have no prefix.
- sync-file in backing-image-manager, but that make the action name there not consistent.
- UI
- Harvester side

Reply:
I prefer not to change it this time,
- "upload" is for proxy, and we remain the same action name all the way down the proxy path
- "BackingImageCleanup", "BackupBackingImageCreate" follow the convention in "model.go" to have prefix, besides, we also use the same name for their handler.
```
	backingImageActions := map[string]func(http.ResponseWriter, *http.Request) error{
		"backingImageCleanup":      s.BackingImageCleanup,
		BackingImageUpload:         s.fwd.Handler(s.fwd.HandleProxyRequestForBackingImageUpload, UploadParametersForBackingImage(s.m), s.BackingImageGet),
		"backupBackingImageCreate": s.BackupBackingImageCreate, 
        }
```

https://github.com/longhorn/longhorn-manager/pull/2658/files#diff-cb740c237beef9090d8b3a15ca8b76aac83a3af2d52deadc382b4309916a715cR165-R167

Reply:
Discussed. Go with your current implementation.